book/src/dev/rfcs/xxxx-block-subsidy.md

```diff
@@ -0,0 +1,302 @@
+
+https://zips.z.cash/protocol/canopy.pdf#joinsplitencodingandconsensus
+
+The balance rule is that this pool must have non-negative value, and its net value is the miner fee for the transaction.
+
+**Note:** To compute the transparent value pool we need blockchain state. The `tx_in` is always a reference to a previous transaction output, this is different from Sprout and Sapling pools where everything we need is in the same transaction. The details about how this is going to be implemented are outside of the scope of this RFC. They will be documented in a separate contextual validation RFC.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+In Zebra the consensus related code lives in the `zebra-consensus` crate. The block subsidy checks are implemented in `zebra-consensus`, in the following module:
+
+- `zebra-consensus/src/block/subsidy/subsidy.rs` 
+
+Inside `zebra-consensus/src/block/subsidy/` the following submodules will be created:
+
+- `general.rs`: General block reward functions and utilities.
+- `founders_reward.rs`: Specific functions related to funders reward.
+- `funding_streams.rs`: Specific functions for funding streams.
+
+In addition to calculations the block subsidy requires constants defined in the protocol. The implementation will also create additional constants, all of them will live at:
+
+- `zebra-consensus/src/parameters/subsidy.rs`
+
+Checking functions for blocks are implemented at `zebra-consensus/src/block/check.rs` and they are called at `zebra-consensus/src/block.rs`. This follows the already existing structure for block validation in Zebra.
+
+It is important to note that all blocks before Sapling are verified by the CheckpointVerifier, so they are not considered in this design. The following table will show what periods are included and what is not in this proposal:
+
+| Height                                 | Miner    | Founder reward | Funding streams | Shielded Coinbase | Target Spacing |
+|----------------------------------------|----------|----------------|-----------------|-------------------|----------------|             
+| ~Genesis~                              | ~**0%**~ | ~**0%**~       | ~**0%**~        | ~**No**~          | ~**None**~     |
+| ~(Genesis + 1)..Sapling~               | ~80%~    | ~20%~          | ~0%~            | ~No~              | ~150 seconds~  |
+| Sapling..Blossom                       | 80%      | 20%            | 0%              | No                | 150 seconds    |
+| Blossom..Heartwood                     | 80%      | 20%            | 0%              | No                | 75 seconds     |
+| Heartwood..Canopy                      | 80%      | 20%            | 0%              | Yes               | 75 seconds     |
+| Canopy..Second Halving                 | 80%      | 0%             | 20%             | Yes               | 75 seconds     |
```

Comment:
In regards to block subsidy orchard will not change anything as far as i know but should we replace this with `Canopy..Orchad` and add a new row `Orchad..Second Halving` ?

Reply:
We skip network upgrades where nothing changes, like Overwinter. Currently, there are no block subsidy, coinbase or target spacing changes scheduled for NU5.

(Just a reminder, NU5 is the name of the next network upgrade. Orchard is the name of a transaction proving system change that will probably be in NU5.)